{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "import requests\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"biorxiv.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Affiliatin_nums: Replace NAs with 1, for single affiliation.\n",
    "#Affiliation_nums: Split into list\n",
    "df.affiliation_nums = df.affiliation_nums.fillna('1')\n",
    "df.affiliation_nums = df.affiliation_nums.apply(str.split,sep=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Affiliation_text: Replace NAs with empty string\n",
    "#Run clean_affiliation, to reformat affiliations\n",
    "def clean_affiliation(x):\n",
    "    x = x.split(';')\n",
    "    for i in range(len(x)):\n",
    "        x[i] = re.sub('(\\\\n)(\\\\t)+','',x[i])\n",
    "        x[i] = x[i].strip()\n",
    "        x[i] = re.sub('^,+','',x[i])\n",
    "        x[i] = x[i].strip()\n",
    "    return(x)\n",
    "df.affiliation_text = df.affiliation_text.fillna('')\n",
    "df.affiliation_text = df.affiliation_text.apply(clean_affiliation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Cleaning number of downloads\n",
    "def clean_downloads(x):\n",
    "    x = x.split(',')\n",
    "    x = map(lambda y: re.sub('<tr class=\"odd\">','',y),x)\n",
    "    x = map(lambda y: re.sub('<tr class=\"even\">','',y),x)\n",
    "    x = map(lambda y: re.sub(' </tr>','',y),x)\n",
    "    x = list(map(lambda y: y.split('</td>'),x))\n",
    "    for i in range(len(x)):\n",
    "        x[i] = list(map(lambda y: re.sub('^<td>','',y),x[i]))\n",
    "    return x\n",
    "df.downloads = df.downloads.fillna('')\n",
    "df.downloads = df.downloads.apply(clean_downloads)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Cleaning date_posted, converting to datetime\n",
    "tmp = df.date_posted\n",
    "tmp = list(map(lambda x: x[12:],tmp))\n",
    "tmp = list(map(lambda x: x[:-3],tmp))\n",
    "df.date_posted = pd.Series(tmp)\n",
    "df.date_posted = pd.to_datetime(df.date_posted)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_pubmed_id(authors):\n",
    "    base = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/'\n",
    "    esearch= 'esearch.fcgi?'\n",
    "    db= 'db=pubmed&'\n",
    "    field= 'field=author&'\n",
    "    term_prefix = 'term='\n",
    "    and_syntax = '+AND+'\n",
    "    term_suffix = '&'\n",
    "    api = 'api_key=56a5fd6220202c3f7c1c924f215c4188b708'\n",
    "    \n",
    "    if authors=='':\n",
    "        return \"No authors\"\n",
    "    authors = authors.lower()\n",
    "    authors = authors.split(',')\n",
    "    \n",
    "    if len(authors) > 2:\n",
    "        term = re.sub(' ','+',authors[0])+and_syntax+re.sub(' ','+',authors[1])+and_syntax+re.sub(' ','+',authors[2])\n",
    "    elif len(authors) > 1:\n",
    "        term = re.sub(' ','+',authors[0])+and_syntax+re.sub(' ','+',authors[1])\n",
    "    else:\n",
    "        term = re.sub(' ','+',authors[0])\n",
    "        \n",
    "    url = base+esearch+db+field+term_prefix+term+term_suffix+api\n",
    "    \n",
    "    response=requests.get(url)\n",
    "    \n",
    "    return response.text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_pmid(xml):\n",
    "    if xml==\"No authors\":\n",
    "        return [xml]\n",
    "    test_if_error =  re.search('error',xml[:9])\n",
    "    if (test_if_error!=None):\n",
    "        return [xml]\n",
    "    tmp = re.split('<Count>',xml)\n",
    "    tmp = re.split('</Count>',tmp[1])\n",
    "    count = int(tmp[0])\n",
    "    if count ==0:\n",
    "        return [count,\"\"]\n",
    "    else:\n",
    "        tmp = re.split('<IdList>',xml)\n",
    "        tmp = re.split('</IdList>',tmp[1])\n",
    "        tmp = re.split('</Id>',tmp[0])\n",
    "        tmp = map(lambda x: re.sub('<Id>','',x),tmp)\n",
    "        tmp = list(map(lambda x: re.sub('\\n','',x),tmp))\n",
    "        return [count,*tmp]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Obtain authors from the data frame and query Entrez for PMIDs\n",
    "author_list = list(df.author_names.fillna('').values)\n",
    "pmid_result = []\n",
    "for i in range(len(author_list)):\n",
    "    tmp = get_pubmed_id(author_list[i])\n",
    "    pmid_result.append(tmp)\n",
    "    time.sleep(0.2)\n",
    "    \n",
    "#Optionally write result to CSV\n",
    "#tmp = pd.Series(pmid_result)\n",
    "#tmp.to_csv('pmids.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Parse raw PMID returns into count of hits, and PMID numbers. Add as new column to df\n",
    "parsed_pmid = list(map(parse_pmid,pmid_result))\n",
    "df['pmid']=pd.Series(parsed_pmid)\n",
    "\n",
    "#Subset the data on just results that have one hit for a PMID, meaning a one to one mapping of authors to PMID\n",
    "#Assign values to dataframe called df_subset\n",
    "subset_vec = []\n",
    "for i in range(len(df.pmid)):\n",
    "    if len(df.pmid[i])==3:\n",
    "        subset_vec.append(True)\n",
    "    else:\n",
    "        subset_vec.append(False)\n",
    "subset_indices = df.pmid[subset_vec].index\n",
    "df_subset = df.iloc[subset_indices]\n",
    "df_subset.index = range(1432)\n",
    "df_subset.pmid = df_subset.pmid.apply(lambda x: x[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_pmid_links(pmid_list):\n",
    "    \n",
    "    base = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/'\n",
    "    elink= 'elink.fcgi?'\n",
    "    db= 'dbfrom=pubmed&linkname=pubmed_pubmed_citedin&'\n",
    "    id_prefix = 'id='\n",
    "    id_suffix = '&'\n",
    "    api = 'api_key=56a5fd6220202c3f7c1c924f215c4188b708'\n",
    "    \n",
    "    result = []\n",
    "    \n",
    "    sub_query = ''\n",
    "    for i in range(len(pmid_list)):\n",
    "        sub_query = sub_query + id_prefix + pmid_list[i] + id_suffix\n",
    "    url = base+elink+db+sub_query+api\n",
    "    response=requests.get(url)\n",
    "    queries = re.split('<LinkSet>',response.text)[1:]\n",
    "    for i in range(len(queries)):\n",
    "        if re.search('<LinkSetDb>',queries[i])==None:\n",
    "            result.append([])\n",
    "        else:\n",
    "            links = re.split('<Link>\\n\\t\\t\\t\\t<Id>',queries[i])[1:]\n",
    "            links = list(map(lambda x: x[:8],links))\n",
    "            result.append(links)\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#From subsetted data, query the Entrez linking database to get number of citations per paper\n",
    "pmid_list = list(df_subset.pmid.values)\n",
    "link_list = []\n",
    "for i in range(14):\n",
    "    j = 0 + (100*i)\n",
    "    k = 100 + (100*i)\n",
    "    link_list.append(get_pmid_links(pmid_list[j:k]))\n",
    "    time.sleep(1)\n",
    "link_list.append(get_pmid_links(pmid_list[1400:1432]))\n",
    "link_list_tmp =[]\n",
    "for i in range(len(link_list)):\n",
    "    for j in range(len(link_list[i])):\n",
    "        link_list_tmp.append(link_list[i][j])\n",
    "df_subset['pmid_links'] = pd.Series(link_list_tmp)\n",
    "df_subset['pmid_links_num'] = df_subset.pmid_links.apply(len)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_total_downloads(paper):\n",
    "    result = 0\n",
    "    if paper[0]==['']:\n",
    "        return result\n",
    "    for i in range(len(paper)):\n",
    "        result += int(paper[i][1])\n",
    "    return result\n",
    "df_subset['total_downloads']=df_subset.downloads.apply(get_total_downloads)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_subset.to_csv('biorxiv-cleaned.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>affiliation_nums</th>\n",
       "      <th>affiliation_text</th>\n",
       "      <th>author_names</th>\n",
       "      <th>date_posted</th>\n",
       "      <th>doi</th>\n",
       "      <th>downloads</th>\n",
       "      <th>paper_title</th>\n",
       "      <th>subject_area</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>\\n\\t\\t\\t\\t, University of Southern California\\...</td>\n",
       "      <td>James H Joly,Brandon TL Chew,Nicholas Alexande...</td>\n",
       "      <td>\\n    Posted October 07, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.10.05.327429</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Oct 2020&lt;/td&gt;&lt;td&gt;500&lt;/td&gt;&lt;...</td>\n",
       "      <td>The landscape of metabolic pathway dependencie...</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1,3,4,1,2,3,2,3,2,3,2,2,3,2,3,2,3,1,3,*</td>\n",
       "      <td>, Glasgow G61 1QH, United Kingdom,, Glasgow G6...</td>\n",
       "      <td>Mate Naszai,Yachuan Yu,Alvaro R Fernandez,Emma...</td>\n",
       "      <td>\\n    Posted October 08, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.10.07.329607</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Oct 2020&lt;/td&gt;&lt;td&gt;343&lt;/td&gt;&lt;...</td>\n",
       "      <td>RAL GTPases mediate EGFR/MAPK signalling-drive...</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1,4,2,3,3,2,2</td>\n",
       "      <td>\\n\\t\\t\\t\\t, H. Lee Moffitt Cancer Center &amp; Res...</td>\n",
       "      <td>Jeffrey West,Maximilian Strobl,Cole Armagost,R...</td>\n",
       "      <td>\\n    Posted October 09, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.10.08.331678</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Oct 2020&lt;/td&gt;&lt;td&gt;280&lt;/td&gt;&lt;...</td>\n",
       "      <td>Antifragile therapy</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1,10,1,1,1,1,2,1,1,1,3,3,4,4,4,5,1,3,3,3,3,3,1...</td>\n",
       "      <td>\\n\\t\\t\\t\\t, University of Cambridge;\\n\\t\\t\\t,\\...</td>\n",
       "      <td>Filipe Correia Martins,Dominique-Laurent Coutu...</td>\n",
       "      <td>\\n    Posted October 12, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.10.04.325365</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Oct 2020&lt;/td&gt;&lt;td&gt;768&lt;/td&gt;&lt;...</td>\n",
       "      <td>Somatic chromosomal number alterations affecti...</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1,2,3,1,4,5,4,6,7,1,9,8</td>\n",
       "      <td>\\n\\t\\t\\t\\t, Helen Diller Comprehensive Cancer ...</td>\n",
       "      <td>Andrew L Wolfe,Qingwen Zhou,Eneda Toska,Jacque...</td>\n",
       "      <td>\\n    Posted October 13, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.10.13.337998</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Oct 2020&lt;/td&gt;&lt;td&gt;353&lt;/td&gt;&lt;...</td>\n",
       "      <td>UDP-glucose pyrophosphorylase 2, a regulator o...</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3634</th>\n",
       "      <td>1,+,#,1,2,3,4,2,5,1,*</td>\n",
       "      <td>, Tucson, Arizona,, Tucson, Arizona,, Tucson, ...</td>\n",
       "      <td>Brenna A. Rheinheimer,Ronald L. Heimark,Adam D...</td>\n",
       "      <td>\\n    Posted September 30, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.09.29.318873</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Sep 2020&lt;/td&gt;&lt;td&gt;194&lt;/td&gt;&lt;...</td>\n",
       "      <td>Cell intrinsic signaling in , mutant pancreati...</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3635</th>\n",
       "      <td>1,2,1,1,1,3,4,1,1,1,1,1,*,5,6,*,1,*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lei Huang,Xiao-Ou Zhang,Odette Verdejo-Torres,...</td>\n",
       "      <td>\\n    Posted September 30, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.08.12.246660</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Aug 2020&lt;/td&gt;&lt;td&gt;703&lt;/td&gt;&lt;...</td>\n",
       "      <td>Protein arginine methyltransferase 5 promotes ...</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3636</th>\n",
       "      <td>1,5,2,5,2,5,2,1,5,3,1,4,1,5,*</td>\n",
       "      <td>, Houston TX,, Houston TX,, Houston TX,, Houst...</td>\n",
       "      <td>Emilly S Villodre,Yun Gong,Lei Huo,Esther C Yo...</td>\n",
       "      <td>\\n    Posted September 28, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.09.25.313817</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Sep 2020&lt;/td&gt;&lt;td&gt;372&lt;/td&gt;&lt;...</td>\n",
       "      <td>NDRG1 expression is an independent prognostic ...</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3637</th>\n",
       "      <td>1,#,1,#,2,1,3,3,3,3,3,*,1,*</td>\n",
       "      <td>, Beijing 100037, China,, Chengdu 610041, Chin...</td>\n",
       "      <td>Xuanyu Liu,Meng Yuan,Qinqin Xiang,Wen Chen,Zhu...</td>\n",
       "      <td>\\n    Posted September 28, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.09.27.315911</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Sep 2020&lt;/td&gt;&lt;td&gt;248&lt;/td&gt;&lt;...</td>\n",
       "      <td>Single-cell RNA-seq of the stromal vascular fr...</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3638</th>\n",
       "      <td>1,2,3,1,1,2,3,4,*</td>\n",
       "      <td>, Auckland, New Zealand, Auckland, New Zealand...</td>\n",
       "      <td>Colin L. Hisey,Petr Tomek,Yohanes N.S. Nursali...</td>\n",
       "      <td>\\n    Posted September 28, 2020.</td>\n",
       "      <td>https://doi.org/10.1101/2020.09.27.309252</td>\n",
       "      <td>&lt;tr class=\"odd\"&gt;&lt;td&gt;Sep 2020&lt;/td&gt;&lt;td&gt;223&lt;/td&gt;&lt;...</td>\n",
       "      <td>Towards establishing Extracellular Vesicle-ass...</td>\n",
       "      <td>Cancer Biology,\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3639 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       affiliation_nums  \\\n",
       "0                                                     1   \n",
       "1               1,3,4,1,2,3,2,3,2,3,2,2,3,2,3,2,3,1,3,*   \n",
       "2                                         1,4,2,3,3,2,2   \n",
       "3     1,10,1,1,1,1,2,1,1,1,3,3,4,4,4,5,1,3,3,3,3,3,1...   \n",
       "4                               1,2,3,1,4,5,4,6,7,1,9,8   \n",
       "...                                                 ...   \n",
       "3634                              1,+,#,1,2,3,4,2,5,1,*   \n",
       "3635                1,2,1,1,1,3,4,1,1,1,1,1,*,5,6,*,1,*   \n",
       "3636                      1,5,2,5,2,5,2,1,5,3,1,4,1,5,*   \n",
       "3637                        1,#,1,#,2,1,3,3,3,3,3,*,1,*   \n",
       "3638                                  1,2,3,1,1,2,3,4,*   \n",
       "\n",
       "                                       affiliation_text  \\\n",
       "0     \\n\\t\\t\\t\\t, University of Southern California\\...   \n",
       "1     , Glasgow G61 1QH, United Kingdom,, Glasgow G6...   \n",
       "2     \\n\\t\\t\\t\\t, H. Lee Moffitt Cancer Center & Res...   \n",
       "3     \\n\\t\\t\\t\\t, University of Cambridge;\\n\\t\\t\\t,\\...   \n",
       "4     \\n\\t\\t\\t\\t, Helen Diller Comprehensive Cancer ...   \n",
       "...                                                 ...   \n",
       "3634  , Tucson, Arizona,, Tucson, Arizona,, Tucson, ...   \n",
       "3635                                                NaN   \n",
       "3636  , Houston TX,, Houston TX,, Houston TX,, Houst...   \n",
       "3637  , Beijing 100037, China,, Chengdu 610041, Chin...   \n",
       "3638  , Auckland, New Zealand, Auckland, New Zealand...   \n",
       "\n",
       "                                           author_names  \\\n",
       "0     James H Joly,Brandon TL Chew,Nicholas Alexande...   \n",
       "1     Mate Naszai,Yachuan Yu,Alvaro R Fernandez,Emma...   \n",
       "2     Jeffrey West,Maximilian Strobl,Cole Armagost,R...   \n",
       "3     Filipe Correia Martins,Dominique-Laurent Coutu...   \n",
       "4     Andrew L Wolfe,Qingwen Zhou,Eneda Toska,Jacque...   \n",
       "...                                                 ...   \n",
       "3634  Brenna A. Rheinheimer,Ronald L. Heimark,Adam D...   \n",
       "3635  Lei Huang,Xiao-Ou Zhang,Odette Verdejo-Torres,...   \n",
       "3636  Emilly S Villodre,Yun Gong,Lei Huo,Esther C Yo...   \n",
       "3637  Xuanyu Liu,Meng Yuan,Qinqin Xiang,Wen Chen,Zhu...   \n",
       "3638  Colin L. Hisey,Petr Tomek,Yohanes N.S. Nursali...   \n",
       "\n",
       "                             date_posted  \\\n",
       "0       \\n    Posted October 07, 2020.     \n",
       "1       \\n    Posted October 08, 2020.     \n",
       "2       \\n    Posted October 09, 2020.     \n",
       "3       \\n    Posted October 12, 2020.     \n",
       "4       \\n    Posted October 13, 2020.     \n",
       "...                                  ...   \n",
       "3634  \\n    Posted September 30, 2020.     \n",
       "3635  \\n    Posted September 30, 2020.     \n",
       "3636  \\n    Posted September 28, 2020.     \n",
       "3637  \\n    Posted September 28, 2020.     \n",
       "3638  \\n    Posted September 28, 2020.     \n",
       "\n",
       "                                              doi  \\\n",
       "0      https://doi.org/10.1101/2020.10.05.327429    \n",
       "1      https://doi.org/10.1101/2020.10.07.329607    \n",
       "2      https://doi.org/10.1101/2020.10.08.331678    \n",
       "3      https://doi.org/10.1101/2020.10.04.325365    \n",
       "4      https://doi.org/10.1101/2020.10.13.337998    \n",
       "...                                           ...   \n",
       "3634   https://doi.org/10.1101/2020.09.29.318873    \n",
       "3635   https://doi.org/10.1101/2020.08.12.246660    \n",
       "3636   https://doi.org/10.1101/2020.09.25.313817    \n",
       "3637   https://doi.org/10.1101/2020.09.27.315911    \n",
       "3638   https://doi.org/10.1101/2020.09.27.309252    \n",
       "\n",
       "                                              downloads  \\\n",
       "0     <tr class=\"odd\"><td>Oct 2020</td><td>500</td><...   \n",
       "1     <tr class=\"odd\"><td>Oct 2020</td><td>343</td><...   \n",
       "2     <tr class=\"odd\"><td>Oct 2020</td><td>280</td><...   \n",
       "3     <tr class=\"odd\"><td>Oct 2020</td><td>768</td><...   \n",
       "4     <tr class=\"odd\"><td>Oct 2020</td><td>353</td><...   \n",
       "...                                                 ...   \n",
       "3634  <tr class=\"odd\"><td>Sep 2020</td><td>194</td><...   \n",
       "3635  <tr class=\"odd\"><td>Aug 2020</td><td>703</td><...   \n",
       "3636  <tr class=\"odd\"><td>Sep 2020</td><td>372</td><...   \n",
       "3637  <tr class=\"odd\"><td>Sep 2020</td><td>248</td><...   \n",
       "3638  <tr class=\"odd\"><td>Sep 2020</td><td>223</td><...   \n",
       "\n",
       "                                            paper_title       subject_area  \n",
       "0     The landscape of metabolic pathway dependencie...  Cancer Biology,\\n  \n",
       "1     RAL GTPases mediate EGFR/MAPK signalling-drive...  Cancer Biology,\\n  \n",
       "2                                   Antifragile therapy  Cancer Biology,\\n  \n",
       "3     Somatic chromosomal number alterations affecti...  Cancer Biology,\\n  \n",
       "4     UDP-glucose pyrophosphorylase 2, a regulator o...  Cancer Biology,\\n  \n",
       "...                                                 ...                ...  \n",
       "3634  Cell intrinsic signaling in , mutant pancreati...  Cancer Biology,\\n  \n",
       "3635  Protein arginine methyltransferase 5 promotes ...  Cancer Biology,\\n  \n",
       "3636  NDRG1 expression is an independent prognostic ...  Cancer Biology,\\n  \n",
       "3637  Single-cell RNA-seq of the stromal vascular fr...  Cancer Biology,\\n  \n",
       "3638  Towards establishing Extracellular Vesicle-ass...  Cancer Biology,\\n  \n",
       "\n",
       "[3639 rows x 8 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
